web/src/app/converters/submission-data-converter.ts

```diff
@@ -229,30 +229,42 @@ export class LegacySubmissionDataConverter {
   }
 
   private static toResult(
-    resultValue: number | string | List<string>,
+    resultValue:
+      | number
+      | string
+      | string[]
+      | Timestamp
+      | {type?: string; geometry?: {type: string}},
     task?: Task
   ): Result | Error {
     try {
-      if (typeof resultValue === 'string') {
-        return new Result(resultValue as string);
-      } else if (typeof resultValue === 'number') {
+      if (typeof resultValue === 'number') {
         return new Result(resultValue as number);
+      } else if (typeof resultValue === 'string') {
+        return new Result(resultValue as string);
       } else if (resultValue instanceof Array) {
         return new Result(
           List(
-            resultValue
-              .filter(optionId => !optionId.startsWith('['))
-              .map(
-                optionId =>
+            resultValue.map(optionId => {
+              if (optionId.startsWith('['))
+                return new Option(
+                  'otherOption',
+                  'otherOption',
```

Comment:
Where do these special values `'otherOption'` come from? Can we refer to a const w/docs which explain their usage?